Migrate hooks api test to playwright

pooja-muchandikar opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->
## What?
<!-- In a few words, what is the PR actually doing? -->
Part of https://github.com/WordPress/gutenberg/issues/38851.
Migrate[ hooks-api.test.js ](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/plugins/hooks-api.test.js) to its Playwright version.
## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Part of https://github.com/WordPress/gutenberg/issues/38851.
## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
See[ MIGRATION.md](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) for migration steps.
## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->
Run` npm run test-e2e:playwright test/e2e/specs/editor/plugins/hooks-api.spec.js`

pooja-muchandikar committed:
Migrate hooks api test to playwright

pooja-muchandikar commented:
Hi @kevin940726 👋 

I hope you are doing great!!

All CI checks have passed for this PR except the `React Native E2E Tests (iOS) / test` CI check.  It is failed but seems like its not related to the test case I worked on.. 

![image](https://user-images.githubusercontent.com/41000648/180165740-9124b01b-ad34-49fb-b395-bc099aa67a2e.png)

Could you please help me in reviewing this PR?

Thanks!!

talldan commented on test/e2e/specs/editor/plugins/hooks-api.spec.js:
```diff
@@ -0,0 +1,44 @@
+		await page.keyboard.type( 'First paragraph' );
+		expect(
+			page.locator( '.edit-post-sidebar .e2e-reset-block-button' )
+		).not.toBeNull();
```
I think it'd be good to use playwright's `toBeVisible()` instead of `not.toBeNull()`.

talldan commented on test/e2e/specs/editor/plugins/hooks-api.spec.js:
```diff
@@ -0,0 +1,44 @@
+			'First paragraph'
+		);
+		await page.click( 'role=button[name="Reset Block"i]' );
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
```
The snapshot is an empty string, so I think it's ok to change this to 
```js
expect( await editor.getEditedPostContent() ).toEqual( '' );
```

And then remove the snapshot file (`test/e2e/specs/editor/plugins/__snapshots__/Using-Hooks-API-Pressing-reset-block-button-resets-the-block-1-chromium.txt`)

talldan commented on test/e2e/specs/editor/plugins/hooks-api.spec.js:
```diff
@@ -0,0 +1,44 @@
+		const content = page.locator( 'p[data-type="core/paragraph"]' );
+		expect( await content.evaluate( ( node ) => node.innerText ) ).toEqual(
+			'First paragraph'
+		);
```
Some of the playwright APIs should make this a bit simpler to write:
```suggestion
		const paragraphBlock = page.locator( 'role=document[name="Paragraph block"i]' );
		await expect( paragraphBlock ).toHaveText( 'First paragraph' );
```

I haven't tested that the above code works btw, but hopefully it sends you down the right path.

talldan commented on test/e2e/specs/editor/plugins/hooks-api.spec.js:
```diff
@@ -0,0 +1,44 @@
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( 'First paragraph' );
+		expect(
+			page.locator( '.edit-post-sidebar .e2e-reset-block-button' )
```
Lower down in the code (line 41) the locator `'role=button[name="Reset Block"i]'` is being used, and it'd be great to use that here as well instead of the classnames.

talldan reviewed (commented):
Thanks for working on this.

It looks good so far, but there are a few small improvement that I think can be made.

pooja-muchandikar committed:
Address review feedbacks

pooja-muchandikar commented:
Hi @talldan,

I have addressed all the feedbacks and pushed the changes. Also the CI is passed :) 

Please take a look and let me know if any further changes required. 

Thanks!

talldan reviewed (approved):
Looks good, thanks for addressing those changes!
